### PR TITLE
Increase unit test coverage

### LINKED ISSUE #515: Increase MSF4J unit test coverage
**Description:**
Increase MSF4J unit test coverage for following components:

- org.wso2.msf4j.formparam
- org.wso2.msf4j.formparam.util
- org.wso2.msf4j.formparam.util.mime
- org.wso2.msf4j.interceptor
- org.wso2.msf4j.security
- org.wso2.msf4j.util

### PR DESCRIPTION
**Pull request by vidurananayakkara**
## Purpose
Resolves #515 

Increase MSF4J unit test coverage for following components:

- org.wso2.msf4j.formparam
- org.wso2.msf4j.formparam.util
- org.wso2.msf4j.formparam.util.mime
- org.wso2.msf4j.interceptor
- org.wso2.msf4j.security
- org.wso2.msf4j.util

The before/after test coverage are as described in the Automation tests section.

## Goals
Increase MSF4J unit test coverage for following components:

- org.wso2.msf4j.formparam
- org.wso2.msf4j.formparam.util
- org.wso2.msf4j.formparam.util.mime
- org.wso2.msf4j.interceptor
- org.wso2.msf4j.security
- org.wso2.msf4j.util

## Release note
* Increased unit test coverage

## Automation tests
 - Unit tests 

|                                    | **Old**           | **New**            | **Old**            | **New**            | **Old**             | **New**             |
|------------------------------------|---------------|----------------|----------------|----------------|-----------------|-----------------|
| **Element**                            | **Class, %**      | **Class, %**       | **Method, %**      | **Method, %**      | **Line, %**         | **Line, %**         |
| org.wso2.msf4j                     | 63% (17/ 27)  | 63.3% (19/ 30) | 8.4% (25/ 299) | 8.6% (27/ 315) | 3.5% (65/ 1882) | 3.3% (67/ 2028) |
| org.wso2.msf4j.beanconversion      | 33.3% (1/ 3)  | 33.3% (1/ 3)   | 7.7% (1/ 13)   | 6.2% (1/ 16)   | 1.6% (1/ 63)    | 1.4% (1/ 71)    |
| org.wso2.msf4j.config              | 100% (1/ 1)   | 100% (2/ 2)    | 60% (3/ 5)     | 57.1% (4/ 7)   | 55.6% (5/ 9)    | 35.3% (6/ 17)   |
| org.wso2.msf4j.example             | 0% (0/ 27)    | 0% (0/ 27)     | 0% (0/ 134)    | 0% (0/ 134)    | 0% (0/ 358)     | 0% (0/ 361)     |
| org.wso2.msf4j.formparam           | 0% (0/ 10)    | 9.1% (1/ 11)   | 0% (0/ 77)     | 1.3% (1/ 79)   | 0% (0/ 466)     | 0.2% (1/ 484)   |
| org.wso2.msf4j.formparam.util      | 0% (0/ 2)     | 33.3% (1/ 3)   | 0% (0/ 11)     | 6.2% (1/ 16)   | 0% (0/ 57)      | 1.3% (1/ 78)    |
| org.wso2.msf4j.formparam.util.mime | 0% (0/ 4)     | 42.9% (3/ 7)   | 0% (0/ 12)     | 13% (3/ 23)    | 0% (0/ 153)     | 1.5% (3/ 196)   |
| org.wso2.msf4j.interceptor         | 6.7% (2/ 30)  | 9.7% (3/ 31)   | 2.5% (3/ 118)  | 3.3% (4/ 121)  | 2.3% (6/ 261)   | 2.5% (7/ 280)   |
| org.wso2.msf4j.internal            | 60% (6/ 10)   | 60% (6/ 10)    | 12% (15/ 125)  | 12% (15/ 125)  | 9% (61/ 677)    | 9% (61/ 676)    |
| org.wso2.msf4j.security            | 0% (0/ 3)     | 42.9% (3/ 7)   | 0% (0/ 11)     | 10.7% (3/ 28)  | 0% (0/ 46)      | 2.4% (3/ 124)   |
| org.wso2.msf4j.service             | 45.5% (5/ 11) | 38.5% (5/ 13)  | 5.4% (6/ 111)  | 5.1% (6/ 118)  | 2.6% (7/ 273)   | 2.3% (7/ 302)   |
| org.wso2.msf4j.util                | 5.9% (1/ 17)  | 15.8% (3/ 19)  | 1.2% (1/ 80)   | 3.6% (3/ 84)   | 0.2% (1/ 502)   | 1.4% (7/ 516)   |

 - Integration tests
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
Java 8u151, Ubuntu 16.04, Maven 3.3.9

### CODE REVIEW

File: core/src/test/java/org/wso2/msf4j/TestUtil.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            urlConn.setDoOutput(true);
+        }
+        urlConn.setRequestMethod(method);
+        if (!keepAlive) {
+            urlConn.setRequestProperty(HEADER_KEY_CONNECTION, HEADER_VAL_CLOSE);
+        }
+
+        return urlConn;
+    }
+
```
Comment (KavinduZoysa) [inline core/src/test/java/org/wso2/msf4j/TestUtil.java]: This method and previous method has duplicated code. Can we move them into one method?

**Comment by Niranjan-K** [comment]
@vidurananayakkara : You need to create an issue for this

File: core/src/test/java/org/wso2/msf4j/TestUtil.java (hunk 1 above)
Comment (vidurananayakkara) [reply core/src/test/java/org/wso2/msf4j/TestUtil.java]: Blank line removed

File: core/src/test/java/org/wso2/msf4j/PersistentSessionManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        content = TestUtil.getContent(urlConnection);
+        assertEquals("3", content);
+        urlConnection.disconnect();
+    }
+
+    @AfterClass
+    public void clean() {
+        microservicesRunner.stop();
+    }
+}
```
Comment (Niranjan-K) [inline core/src/test/java/org/wso2/msf4j/PersistentSessionManagerTest.java]: We need a line break at the end of the file.

File: core/src/test/java/org/wso2/msf4j/ResponseTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    @Test
+    public void testCarbonMessage() {
+        CarbonMessage carbonMessage = response.getCarbonMessage();
+        assertTrue(carbonMessage.isEmpty());
+        assertTrue(carbonMessage.isBufferContent());
+        assertFalse(carbonMessage.isEndOfMsgAdded());
+        assertFalse(carbonMessage.isAlreadyRead());
+        assertFalse(carbonMessage.isFaulty());
+    }
+}
```
Comment (Niranjan-K) [inline core/src/test/java/org/wso2/msf4j/ResponseTest.java]: We need a line break at the end of the file.

File: core/src/test/java/org/wso2/msf4j/TestUtil.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            int ch2 = input2.read();
+            if (ch != ch2) {
+                return false;
+            }
+            ch = input1.read();
+        }
+        int ch2 = input2.read();
+        return (ch2 == -1);
+    }
+}
```
Comment (Niranjan-K) [inline core/src/test/java/org/wso2/msf4j/TestUtil.java]: We need a line break at the end of the file.

File: core/src/test/java/org/wso2/msf4j/config/MSF4JConfigTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    public void testMSF4JConfig() {
+        String threadPoolName = "Test-MSF4J-request-pool";
+        MSF4JConfig msf4JConfig = new MSF4JConfig();
+        msf4JConfig.setThreadCount(100);
+        msf4JConfig.setThreadPoolName(threadPoolName);
+
+        assertEquals(msf4JConfig.getThreadCount(), 100);
+        assertEquals(msf4JConfig.getThreadPoolName(), threadPoolName);
+    }
+}
```
Comment (Niranjan-K) [inline core/src/test/java/org/wso2/msf4j/config/MSF4JConfigTest.java]: We need a line break at the end of the file.

File: core/src/test/java/org/wso2/msf4j/formparam/util/mime/MimeUtilityTest.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+*/
+package org.wso2.msf4j.formparam.util.mime;
+
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import static org.testng.AssertJUnit.assertEquals;
+
```
Comment (Niranjan-K) [inline core/src/test/java/org/wso2/msf4j/formparam/util/mime/MimeUtilityTest.java]: JavaDocs?

File: core/src/test/java/org/wso2/msf4j/PersistentSessionManagerTest.java (hunk 2 above)
Comment (vidurananayakkara) [reply core/src/test/java/org/wso2/msf4j/PersistentSessionManagerTest.java]: Line break is already there at the end of the file

File: core/src/test/java/org/wso2/msf4j/ResponseTest.java (hunk 3 above)
Comment (vidurananayakkara) [reply core/src/test/java/org/wso2/msf4j/ResponseTest.java]: Line break is already there at the end of the file

File: core/src/test/java/org/wso2/msf4j/config/MSF4JConfigTest.java (hunk 5 above)
Comment (vidurananayakkara) [reply core/src/test/java/org/wso2/msf4j/config/MSF4JConfigTest.java]: Line break is already there at the end of the file

File: core/src/test/java/org/wso2/msf4j/TestUtil.java (hunk 4 above)
Comment (vidurananayakkara) [reply core/src/test/java/org/wso2/msf4j/TestUtil.java]: Line break is already there at the end of the file

**Comment by vidurananayakkara** [comment]
Closing this PR since a new PR will be sent with the requested changes

**Pull request closed without merging**